keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/probabilistic/TensorUniformVertex.java

```diff
@@ -0,0 +1,99 @@
+package io.improbable.keanu.vertices.dbltensor.probabilistic;
+
+import io.improbable.keanu.distributions.tensors.continuous.TensorUniform;
+import io.improbable.keanu.vertices.dbltensor.DoubleTensor;
+import io.improbable.keanu.vertices.dbltensor.DoubleTensorVertex;
+import io.improbable.keanu.vertices.dbltensor.KeanuRandom;
+import io.improbable.keanu.vertices.dbltensor.nonprobabilistic.ConstantTensorVertex;
+
+import java.util.Map;
+
+import static io.improbable.keanu.vertices.dbltensor.probabilistic.ProbabilisticVertexShaping.checkParentShapes;
+import static io.improbable.keanu.vertices.dbltensor.probabilistic.ProbabilisticVertexShaping.getShapeProposal;
+import static java.util.Collections.singletonMap;
+
+public class TensorUniformVertex extends ProbabilisticDoubleTensor {
+
+    private final DoubleTensorVertex xMin;
+    private final DoubleTensorVertex xMax;
+    private final KeanuRandom random;
+
+    /**
+     * @param shape  desired tensor shape
+     * @param xMin   inclusive
+     * @param xMax   exclusive
```

Comment:
Ah, this is why we have the `greaterThanMask` not `greaterThanOrEqualTo`!

I didn't know this was inclusive only at the bottom end - why is this?